### ISSUE TITLE
Lint/DuplicateMethods for classes inside `class << self` of parent module

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop#5022]
As in #4758, but with `class << self`.

Real-life case: https://github.com/panthomakos/timezone/blob/b70e1d8/lib/timezone/lookup.rb

--------

## Expected behavior

No warnings.

## Actual behavior

```
Inspecting 1 file
W

Offenses:

.test.rb:10:7: W: Method Foo.config is defined at both .test.rb:4 and .test.rb:10.
      attr_reader :config
      ^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

```ruby
# Module with class-methods
module Foo
  class << self
    def config
      'Foo.config'
    end

    # Private class for module
    class Bar
      attr_reader :config
    end
  end
end
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-linux)
```

**Comment by AlexWayfer** [comment]
The bug is still here.

**Comment by AnatolySt** [comment]
Yes. This bug is still here. Rubocop 0.77.0
